src/main/java/org/broadinstitute/hellbender/Main.java

```diff
@@ -241,16 +247,22 @@ private static boolean printStackTraceOnUserExceptions() {
      * Returns the command line program specified, or prints the usage and exits with exit code 1 *
      */
     private static CommandLineProgram extractCommandLineProgram(final String[] args, final List<String> packageList, final List<Class<? extends CommandLineProgram>> classList, final String commandLineName) {
+
         /** Get the set of classes that are our command line programs **/
         final ClassFinder classFinder = new ClassFinder();
         for (final String pkg : packageList) {
+            classFinder.find(pkg, picard.cmdline.CommandLineProgram.class);
```

Comment:
This is getting really complicated for combining projects/toolkits and re-use this main. I guess that my idea of adding a really simple `CommandLineProgram` interface in Barclay is worth now and something simple to implement (just with a single method `instanceMain(String[] args)` to being able to combine toolkits without this big changes. I will submit a PR to add that simple layout to simplify things.

In addtion, a simple `Main` class could be implemented in Barclay to keep some of this logic and simplify things a bit, but that's a bigger change. I would submit a different PR for this to Barclay.

Reply:
By the way, without the interface in Barclay, downstream projects are unable to specify single Picard classes in `getClassList()`, and it would require yet another method to return picard classess...which is kind of cumbersome...